### PR TITLE
arm64: dts: qcom: msm8953: add spi_3, spi_5 and spi_6 interfaces

### PR DESCRIPTION
**Pull request by M0Rf30**
Signed-off-by: Gianluca Boiano <morf3089@gmail.com

### CODE REVIEW

**Comment by M0Rf30** [comment]
Following this
https://github.com/msm8953-mainline/linux/pull/95
hereby I only enable SPI interfaces

File: arch/arm64/boot/dts/qcom/msm8953.dtsi [hunk 1]
```diff
@@ -2247,6 +2275,23 @@
 			status = "disabled";
 		};
 
+		spi_3: spi@78b7000 {
+			compatible = "qcom,spi-qup-v2.2.1";
+			reg = <0x78b7000 0x600>;
+			interrupts = <GIC_SPI 97 IRQ_TYPE_LEVEL_HIGH>;
+			clock-names = "core", "iface";
+			clocks = <&gcc GCC_BLSP1_QUP3_I2C_APPS_CLK>,
+				 <&gcc GCC_BLSP1_AHB_CLK>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/msm8953.dtsi:2284]: Can probably add (with #87 as dependency)
```c
dmas = <&blsp1_dma 8>, <&blsp1_dma 9>;
dma-names = "tx", "rx";
```

File: arch/arm64/boot/dts/qcom/msm8953.dtsi [hunk 2]
```diff
@@ -2298,6 +2343,23 @@
 			status = "disabled";
 		};
 
+		spi_6: spi@7af6000 {
+			compatible = "qcom,spi-qup-v2.2.1";
+			reg = <0x7af6000 0x600>;
+			interrupts = <GIC_SPI 300 IRQ_TYPE_LEVEL_HIGH>;
+			clock-names = "core", "iface";
+			clocks = <&gcc GCC_BLSP2_QUP2_I2C_APPS_CLK>,
+				 <&gcc GCC_BLSP2_AHB_CLK>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/msm8953.dtsi:2352]: Can probably add (with #87 as dependency)
```c
dmas = <&blsp2_dma 6>, <&blsp2_dma 7>;
dma-names = "tx", "rx";
```

**Comment by z3ntu** [comment]
Please also send this patch upstream, no need to keep it in this repo only.

File: arch/arm64/boot/dts/qcom/msm8953.dtsi (hunk 1 above)
Comment (M0Rf30) [reply arch/arm64/boot/dts/qcom/msm8953.dtsi:2284]: Done

**Comment by barni2000** [comment]
@M0Rf30 Can you add spi_5 also?
Downstream node from ysl sources:
```
spi_5: spi@7af5000 { /* BLSP2 QUP1 */
		compatible = "qcom,spi-qup-v2";
		#address-cells = <1>;
		#size-cells = <0>;
		reg-names = "spi_physical", "spi_bam_physical";
		reg = <0x7af5000 0x600>,
			<0x07AC4000 0x1f000>;
		interrupt-names = "spi_irq", "spi_bam_irq";
		interrupts = <0 299 0>, <0 239 0>;
		spi-max-frequency = <19200000>;
		pinctrl-names = "spi_default", "spi_sleep";
		pinctrl-0 = <&spi5_default &spi5_cs0_active>;
		pinctrl-1 = <&spi5_sleep &spi5_cs0_sleep>;
		clocks = <&clock_gcc clk_gcc_blsp2_ahb_clk>,
			<&clock_gcc clk_gcc_blsp2_qup1_spi_apps_clk>;
		clock-names = "iface_clk", "core_clk";
		qcom,infinite-mode = <0>;
		qcom,use-bam;
		qcom,use-pinctrl;
		qcom,ver-reg-exists;
		qcom,bam-consumer-pipe-index = <4>;
		qcom,bam-producer-pipe-index = <5>;
		qcom,master-id = <84>;
	};
```
https://github.com/MiCode/Xiaomi_Kernel_OpenSource/commit/36039a35bfaf3e0fbfef78b5c70a6aa798e56363

**Comment by M0Rf30** [comment]
> @M0Rf30 Can you add spi_5 also? Downstream node from ysl sources:
> 
> ```
> spi_5: spi@7af5000 { /* BLSP2 QUP1 */
> 		compatible = "qcom,spi-qup-v2";
> 		#address-cells = <1>;
> 		#size-cells = <0>;
> 		reg-names = "spi_physical", "spi_bam_physical";
> 		reg = <0x7af5000 0x600>,
> 			<0x07AC4000 0x1f000>;
> 		interrupt-names = "spi_irq", "spi_bam_irq";
> 		interrupts = <0 299 0>, <0 239 0>;
> 		spi-max-frequency = <19200000>;
> 		pinctrl-names = "spi_default", "spi_sleep";
> 		pinctrl-0 = <&spi5_default &spi5_cs0_active>;
> 		pinctrl-1 = <&spi5_sleep &spi5_cs0_sleep>;
> 		clocks = <&clock_gcc clk_gcc_blsp2_ahb_clk>,
> 			<&clock_gcc clk_gcc_blsp2_qup1_spi_apps_clk>;
> 		clock-names = "iface_clk", "core_clk";
> 		qcom,infinite-mode = <0>;
> 		qcom,use-bam;
> 		qcom,use-pinctrl;
> 		qcom,ver-reg-exists;
> 		qcom,bam-consumer-pipe-index = <4>;
> 		qcom,bam-producer-pipe-index = <5>;
> 		qcom,master-id = <84>;
> 	};
> ```
> 
> [MiCode/Xiaomi_Kernel_OpenSource@36039a3](https://github.com/MiCode/Xiaomi_Kernel_OpenSource/commit/36039a35bfaf3e0fbfef78b5c70a6aa798e56363)

Done

**Pull request closed without merging**